src/Visitor/ParentConnector.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\SemanticVersionChecker\Visitor;
+
+use PhpParser\NodeVisitorAbstract;
+use PhpParser\Node;
+
+/**
+ * Create parent reference for nodes. Parent reference can be found in 'parent' node attribute.
+ */
+class ParentConnector extends NodeVisitorAbstract
```

Comment:
PHPParser already provides this functionality https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/NodeVisitor/ParentConnectingVisitor.php. Please use class from the library.

Reply:
There is no such visitor class in current version (v4.5.0) of nikic/PHP-Parser. It appeared in later version (since v4.7.0)

Reply:
Do we have a reason not to upgrade to the current 4.10.2? It is preferable to maintain dependencies with actual versions and not introduce custom code where solution maintained by a vendor may be used.

Reply:
I don't know. Upgrade?

Reply:
@vkublytskyi fixed